cmd/jaeger/internal/extension/jaegerstorage/extension_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package jaegerstorage
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/extension"
+	noopmetric "go.opentelemetry.io/otel/metric/noop"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+	"go.uber.org/zap"
+
+	memoryCfg "github.com/jaegertracing/jaeger/pkg/memory/config"
+)
+
+type storageHost struct {
+	t                *testing.T
+	storageExtension component.Component
+}
+
+func (host storageHost) GetExtensions() map[component.ID]component.Component {
+	myMap := make(map[component.ID]component.Component)
+	myMap[ID] = host.storageExtension
+	return myMap
+}
+
+func (host storageHost) ReportFatalError(err error) {
+	host.t.Fatal(err)
+}
+
+func (storageHost) GetFactory(_ component.Kind, _ component.Type) component.Factory {
+	return nil
+}
+
+func (storageHost) GetExporters() map[component.DataType]map[component.ID]component.Component {
+	return nil
+}
+
+func TestStartStorageExtensionError(t *testing.T) {
+	ctx := context.Background()
+	const memstoreName = "memstore"
+
+	storageExtension := makeStorageExtension(t, memstoreName)
+
+	host := componenttest.NewNopHost()
+	err := storageExtension.Start(ctx, host)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, storageExtension.Shutdown(ctx)) })
+
+	err = storageExtension.Start(ctx, host)
+	require.Error(t, err)
+	require.EqualError(t, err, fmt.Sprintf("duplicate memory storage name %s", memstoreName))
+}
+
+func TestGetStorageFactoryError(t *testing.T) {
+	ctx := context.Background()
+	const memstoreName = "memstore"
+
+	storageExtension := makeStorageExtension(t, memstoreName)
+
+	host := componenttest.NewNopHost()
+	err := storageExtension.Start(ctx, host)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, storageExtension.Shutdown(ctx)) })
+
+	_, err = GetStorageFactory(memstoreName, host)
+	require.Error(t, err)
+	require.EqualError(t, err, fmt.Sprintf("cannot find extension '%s' (make sure it's defined earlier in the config)", ID))
+}
+
+func TestStorageExtension(t *testing.T) {
+	ctx := context.Background()
+	const memstoreName = "memstore"
+
+	storageExtension := makeStorageExtension(t, memstoreName)
+
+	host := storageHost{t: t, storageExtension: storageExtension}
```

Comment:
Why do you need custom host that simply returns back the extension itself? Would it not work with NopHost?

Reply:
nvm